packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -513,6 +513,50 @@ class StripeHelper {
     });
   }
 
+  /**
+   * Reactivate a givel subscription for a customer
+   * If a customer has an active subscription that is set to cancel at the period end:
+   *  1. Update the subscription to remain active at the period end
+   *  2. Verify that after the update the subscription is still in an active state
+   *    True: return the updated Subscription
+   *    False: throw an error
+   * If the customer does not own the subscription, throw an error
+   *
+   * @param {string} uid
+   * @param {string} email
+   * @param {Subscription[id]} subscriptionId
+   */
+  async reactivateSubscriptionForCustomer(uid, email, subscriptionId) {
+    const hasSubscription = await this.subscriptionForCustomer(
+      uid,
+      email,
+      subscriptionId
+    );
+    if (!hasSubscription) {
+      throw error.unknownSubscription();
+    }
+    const subscription = await this.stripe.subscriptions.update(
+      subscriptionId,
+      {
+        cancel_at_period_end: false,
+      }
+    );
+
+    if (!['active', 'trialing'].includes(subscription.status)) {
+      const err = new Error(
```

Comment:
What conditions cause this? Is a user likely to keep clicking to try and re-activate it and wonder what's wrong?

Reply:
This was in the existing code. Actually, this check should happen before the update call to verify that the subscription should be allowed to be updated. Doing the check after-the-fact only shows that the subscription wasn't eligible for a state change.

Reply:
I've updated to reflect this opinion. Let me know your thoughts.

Reply:
Did you push your update? The typo above is still present as well.

Reply:
I did! It looks like the typo occurred in more than one place 😓 I will fix the place that I missed

Reply:
Ready to go @bbangert

Reply:
The change is that now it verifies that the subscription is in an eligible state prior to re-activation, rather than checking afterwards.